netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -350,6 +351,15 @@ private void createStream(CreateStreamCommand command, final ChannelPromise prom
     final NettyClientStream stream = command.stream();
     final Http2Headers headers = command.headers();
     stream.id(streamId);
+
+    if (connection().goAwayReceived() && streamId > connection().local().lastStreamKnownByPeer()) {
```

Comment:
@ejona86 I believe this comment was on the old code ... can you take another look?